include/zfs_comutil.h

```diff
@@ -41,6 +41,27 @@ extern int zfs_spa_version_map(int zpl_version);
 #define	ZFS_NUM_LEGACY_HISTORY_EVENTS 41
 extern const char *zfs_history_event_names[ZFS_NUM_LEGACY_HISTORY_EVENTS];
 
+/*
+ * Truncating snprintf() without compiler warnings
+ *
+ * GCC 7.1 with "-Wall" will warn in cases where we're not checking the
+ * snprintf() return code when the buffer could be truncated:
+ *
+ *     libzfs_iter.c: In function ‘zfs_iter_bookmarks’:
+ *     libzfs_iter.c:207:40: error: ‘snprintf’ output may be truncated
+ *     before the last format character [-Werror=format-truncation=]
+ *         (void) snprintf(name, sizeof (name), "%s#%s", zhp->zfs_name,
+ *
+ * In many cases we don't care and just want to silently truncate the buffer.
+ *
+ * This hack fools the compiler by making the size volatile, so that it isn't
+ * able to compute the truncation at compile time, and thus no error.
+ */
+#define	SNPRINTF_TRUNC(str, size, ...)	{				\
+	volatile int s = size;						\
+	snprintf(str, s, __VA_ARGS__);					\
```

Comment:
What's the purpose for the volatile?
Should it be fine if you pass in size directly, as it's only used once?